src/simulation.hpp

```diff
@@ -446,17 +448,24 @@ template <typename problem_t> class AMRSimulation : public amrex::AmrCore
 	// tracer particles
 #ifdef AMREX_PARTICLES
 	void InitParticles();	 // create tracer particles
-	void InitCICParticles(); // create CIC particles
+	void InitPhyParticles(); // create PhysicsParticles
 	int do_tracers = 0;
 	int do_cic_particles = 0;
+	int do_rad_particles = 0;
+	int do_cic_rad_particles = 0;
 	std::unique_ptr<amrex::AmrTracerParticleContainer> TracerPC;
 	std::unique_ptr<quokka::CICParticleContainer> CICParticles;
+	std::unique_ptr<quokka::RadParticleContainer<problem_t>> RadParticles;
+	std::unique_ptr<quokka::CICRadParticleContainer<problem_t>> CICRadParticles;
 #endif
 
 	// external objects
 #ifdef AMREX_USE_ASCENT
 	Ascent ascent_;
 #endif
+
+	// Add PhysicsParticleRegister member
+	std::unique_ptr<quokka::PhysicsParticleRegister<problem_t>> particleRegister_;
```

Comment:
Can you make this just `quokka::PhysicsParticleRegister<problem_t>`? (no `unique_ptr`)?

Reply:
I thought it's more memory-safe to define particleRegister_ as unique_ptr, but maybe I'm wrong.

Reply:
It's more memory-safe compared to a raw pointer, but in this case, I don't think it needs to be a pointer at all.